test/geometry/GeoTrackView.test.cc

```diff
@@ -91,11 +92,18 @@ TEST_F(GeoTrackViewHostTest, track_line)
         EXPECT_EQ(false, geo.is_outside()); // leaving World
     }
 
+    try
     {
         // Track from outside edge fails
+        CELER_LOG(info) << "Init a track with a pointer outside work "
+                           "volume...";
         geo = {{24, 0, 0}, {-1, 0, 0}};
         EXPECT_EQ(true, geo.is_outside());
     }
+    catch (const std::exception& e)
+    {
+        // std::cerr<<"caught exception: "<< e.what() << std::endl;
+    }
```

Comment:
You don't want to catch errors and always throw them away. If this isn't debug code left in accidentally, it should probably use the googletest
```c++
FAIL() << e.what();
```
or you can add an `EXPECT_NOTHROW` asssertion.

Reply:
I could not figure out what to do here.

Reply:
@mrguilima Does the code inside `try` throw an exception? Is it supposed to? If so, which call is supposed to throw an exception?
- If it's not supposed to throw, remove the `try/catch`
- If not but it still throws an exception, fix it
- If a particular call is supposed to throw a particular exception, uses `EXPECT_THROW`
- If a particular call *might* throw depending on whether a bug is fixed, or some external condition, then cause the test to fail with a useful message when it throws

Reply:
Ok, I used `EXPECT_THROW()` instead of the try-catch block.

Reply:
OK, this makes sense now. But there are still issues:
- Does the initialization fail even when not compiling with debug assertions (`CELERITAS_DEBUG`)? If so, it maybe should throw a `RuntimeError` instead of a `DebugError`. And I'm not sure how it should be handled on device if that's the case... since an assertion will cause the whole program to crash.
- Regardless of whether the assertion is thrown, is the state guaranteed to be updated so that `geo.is_outside()` is true?
- For googletest, use `EXPECT_TRUE(geo.is_outside());` instead of `EXPECT_EQ(geo.is_outside(), true);`.